### ISSUE TITLE
The val and test set problem in ResNet-12 mini-Imagenet experiment.

### ISSUE THREAD

**Issue by JieHaoL** [chrysts/dsn_fewshot#8]
Thanks for sharing your excellent works！
I have some questions when I reproduce the results of ProtoNet using ResNet-12：
1. In mini-Imagenet  experiments of Resnet12, why you load the val dataset in train.py (line 84) use “phase=test", which I think it should be “phase=val", or the val dataset will be same with the test dataset in test.py (line 91). 
2.  In your code, it seems that the test dataset and val dataset loaded from the pickle file include some base categories (training data). Dose it cause the data leakage problem？

Thank you very much！

**Comment by chrysts** [comment]
Hi JieHaoL,

1. If you train using train and validation sets then phase for val set you can use 'test' directly. In the normal setting, you can just use validation set for val dataset.

2. I do not understand what you mean as data leakage. The test samples from base classes are not used for calculating the accuracy i.e. only samples from novel classes are counted for testing.
